### PR TITLE
feat(parser): support emit_mode

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

The parser part

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features).
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#6042

### CODE REVIEW

**Review by BugenZhao — APPROVED** [review]
LGTM

File: src/sqlparser/tests/sqlparser_common.rs [hunk 1]
```diff
@@ -3038,13 +3046,66 @@ fn parse_create_materialized_view() {
...
+            assert!(!or_replace);
+            assert_eq!(emit_mode, None);
+        }
+        _ => unreachable!(),
+    }
+}
+
+#[test]
+fn parse_create_materialized_view_emit_immediately() {
+    let sql = "CREATE MATERIALIZED VIEW myschema.myview EMIT IMMEDIATELY AS SELECT foo FROM bar";
```
Comment (fuyufjh) [inline src/sqlparser/tests/sqlparser_common.rs:3065]: I thought it should be part of the query i.e. inside the `SELECT ... ` clause. Did we change our mind?

Comment (TennyZhuang) [reply src/sqlparser/tests/sqlparser_common.rs:3065]: If we defined them on `SELECT`, how to define the semantics on subquery or CTE?

Currently, our conclusion is that we should ensure a mview or a sink can only have one emit option, the only way is defined on view.

Comment (fuyufjh) [reply src/sqlparser/tests/sqlparser_common.rs:3065]: https://github.com/risingwavelabs/rfcs/pull/30 defines the semantics in terms of relational algebra (i.e. given input set, what should be the output set), so the semantics is still well-defined for subqueries or CTEs.

**Pull request merged**